### ISSUE TITLE
HireFire endpoint intermittently gives 500

### ISSUE THREAD

**Issue by priyam-maheshwari** [ryanhiebert/hirefire#54]
Celery WorkerProc config is as follows:
```
def make_celery():
    """
    Generate celery instance object
    Returns:
        Celery instance object
    """
    celery_app = Celery('worker',
                        backend='rpc://',
                        include=['worker.tasks'])
    celery_app.config_from_object('configuration.celery_config')

    return celery_app


app = make_celery()

class Worker1Proc(CeleryProc):
    name = 'worker1'
    queues = os.environ.get("WORKER1_QUEUES").split(',')
    simple_queues = True

class Worker2Proc(CeleryProc):
    name = 'worker2'
    queues = os.environ.get("WORKER2_QUEUES").split(',')
    simple_queues = True

class Worker3Proc(CeleryProc):
    name = 'worker3'
    queues = os.environ.get("WORKER3_QUEUES").split(',')
    simple_queues = True
```

HireFire API using Flask is setup like this:
```
import os

from flask import Flask
from hirefire.contrib.flask.blueprint import build_hirefire_blueprint

app = Flask(__name__)
bp = build_hirefire_blueprint(os.environ['HIREFIRE_TOKEN'],
                              ['worker.celery.Worker1Proc',
                               'worker.celery.Worker2Proc',
                               'worker.celery.Worker3Proc'])
app.register_blueprint(bp)
```
An this endpoint intermittently gives : 
"Couldn't fetch job queues. An unknown error occurred. Contact support if this issue persists."

I am using python version 3.6.8
HireFire==0.8
gunicorn==19.9.0
Flask==1.0.2
Flask-RESTful==0.3.7
celery==4.3.0rc1

**Comment by priyam-maheshwari** [comment]
In logs I can see couple of errors like these:
`TypeError: get_inspect_value() missing 1 required positional argument: 'method'`
`TypeError: get_status_task_counts() missing 1 required positional argument: 'status'`

**Comment by amureki** [comment]
Hello!
We started seeing this as well after updating to `HireFire==0.8`.

```
get_status_task_counts() missing 1 required positional argument: 'status'
get_inspect_value() missing 1 required positional argument: 'method'
Somewhere in this piece of code:


        try:
            return super(KeyDefaultDict, self).__missing__(key)
        except TypeError:
            value = self.default_factory(key)
            self[key] = value
            return value
```

HireFire setup did not change since 0.7 and is more or less following setup guide from docs.

Our configuration:
Python 3.6.7
Django 2.1.7
HireFire 0.8

**Comment by ryanhiebert** [comment]
Can you confirm that you were previously on 0.7? It surprises me that the changes in 0.8 could have that effect. Did celery get updated as well?

**Comment by amureki** [comment]
Hey @ryanhiebert 

we did update hirefire on March 11, then the next day we got this issue.
Last time we updated something celery-related was around a week ago - kombu 4.3.0->4.4.0:
https://github.com/celery/kombu/blob/master/Changelog, but I also do not see anything related there.

But actually, I can see that it appeared only once and not more, so it is not critical for our case.

**Comment by ryanhiebert** [comment]
Interesting. I'm using this library in production, but I'm stuck on Celery 3.1 for terrible reasons. I'm not seeing this issue currently, and I'm not sure how to reproduce it. I'd definitely appreciate any further help anyone can give to help solve this issue.

**Comment by ryanhiebert** [comment]
> reduce the frequency

So it didn't eliminate it? That's interesting. Perhaps its not a new issue, and its being exacerbated by the channel that is being opened and closed every time the request is made somehow. Before this version the channels were opened every time, but not closed, causing channels to be left open and be abandoned. I'm still not quite seeing the connection, though.

**Comment by ryanhiebert** [comment]
I'm on Celery 4 now finally, and the latest version of this package as well. Are you still having this issue, @priyam-maheshwari or @amureki?

**Comment by priyam-maheshwari** [comment]
Yes still getting intermittent errors ```
TypeError: get_inspect_value() missing 1 required positional argument: 'method'
TypeError: get_status_task_counts() missing 1 required positional argument: 'status'
```

**Comment by chickahoona** [comment]
We had similar problems. If you are using redis too as result backend, then there is a new celery version 4.4.X that fixes the underlying problem (or greatly reduces the problem).

**Comment by ryanhiebert** [comment]
Given the update above, and that there hasn't been further discussion, I'm closing this issue. Feel free to comment if you're still seeing more issues or have more data to bring to solving the problem.
